


########################################################## 
##########################################################  
############ DNAm EPICv2 Array data analysis #############  
##########################################################  
########################################################## 


library(BiocManager)
install("jokergoo/IlluminaHumanMethylationEPICv2anno.20a1.hg38")
install("jokergoo/IlluminaHumanMethylationEPICv2manifest") 
# https://support.bioconductor.org/p/9155304/
devtools::install_github("jokergoo/IlluminaHumanMethylationEPICv2manifest") #latest version of EPICv2 array annotation (as per november 2024)
library(minfi)
library(devtools)
library(limma)
library(sva, quietly=T)
install("perishky/dmrff")
install("yousefi138/geograbi")
library(dmrff)
library(geograbi)
library(dplyr)


AF_DNA_methylation_dataset <- read.csv("AF_DNAme_dataset_SampleSheet1.csv")
samples_AF <- AF_DNA_methylation_dataset
load("norm.beta_noXY.pc2.Robj")
norm_noXY.beta

# NOTE: The "norm.beta_noXY.pc2.Robj" file containing normalized beta values was generated by Prof. Christopher Bell - performed initial processing and normalization of raw DNAm data
# "norm.beta_noXY.pc2.Robj" is too large to be uploaded to GitHub (even when compressed) - This file is available upon request; contact corresponding author to obtain it. 
# EWAS analysis performed by Prof. Chris Bell as presented in Figure S6a
# The normalized dataset was used here for downstream plotting and DMR analysis with meffil.

# ðŸ“Š Figure S6 (WWOX and LRRC32 DMCs) 

cg23377942_WWOX_betavalues <- norm_noXY.beta["cg23377942", ]
cg23377942_WWOX_betavalues_dfforplot <- data.frame (Sample = colnames(norm_noXY.beta), disease_status = AF_DNA_methylation_dataset$Sample_Group, beta_value = (cg23377942_WWOX_betavalues))

cg23377942_WWOX_betavalues_dfforplot_PLOT <- ggplot(cg23377942_WWOX_betavalues_dfforplot, aes(x = disease_status, y = beta_value, fill=disease_status))  +
  geom_boxplot(colour = "black", width=0.3) +
  geom_dotplot(binaxis = "y", stackdir='center', dotsize = 1) + scale_fill_manual(values = c("#F4997F","#43B9D1") )+
  scale_y_continuous(name = "Î² value") + 
  scale_x_discrete(name = "Disease status") + 
  theme_classic() + ggtitle("cg23377942; WWOX") + 
  theme(plot.title = element_text(hjust = 0.5,face="bold",size =18)) 
cg23377942_WWOX_betavalues_dfforplot_PLOT



cg17970043_LRRC32_betavalues <- norm_noXY.beta["cg17970043", ]
cg17970043_LRRC32_betavalues_dfforplot <- data.frame (Sample = colnames(norm_noXY.beta), disease_status = AF_DNA_methylation_dataset$Sample_Group, beta_value = (cg17970043_LRRC32_betavalues))


cg17970043_LRRC32_betavalues_dfforplot_PLOT <- ggplot(cg17970043_LRRC32_betavalues_dfforplot, aes(x = disease_status, y = beta_value, fill=disease_status))  +
  geom_boxplot(colour = "black", width=0.3) +
  geom_dotplot(binaxis = "y", stackdir='center', dotsize = 1) + scale_fill_manual(values = c("#F4997F","#43B9D1") )+
  scale_y_continuous(name = "Î² value") + 
  scale_x_discrete(name = "Disease status") + 
  theme_classic() + ggtitle("cg17970043; LRRC32") + 
  theme(plot.title = element_text(hjust = 0.5,face="bold",size =18)) 
cg17970043_LRRC32_betavalues_dfforplot_PLOT




####################### ####################### ####################### ####################### ####################### #######################

head(norm_noXY.beta)
head(beta_norm)
colnames(norm_noXY.beta)
colnames(beta_norm)

all(is.finite(norm_noXY.beta))

norm_noXY.beta_TorF <- is.finite(norm_noXY.beta)
write.csv(norm_noXY.beta_TorF, "norm_noXY.beta_TorF.csv") #Realized checking the file that those probe named nv-GRCh38-chr do have infinite numbers --> Remove those rows from your beta values matrix (803 probes)
write.csv(norm_noXY.beta, "norm_noXY.beta.csv")

norm_noXY.beta_rownumbers <-  grep("nv-GRCh38-chr",rownames(norm_noXY.beta))
norm_noXY.beta <- norm_noXY.beta[-norm_noXY.beta_rownumbers,]
dim(norm_noXY.beta) #884510

methylation <- norm_noXY.beta

library(sva, quietly=T)

## construct EWAS model
mod <- model.matrix(~ Sample_Label + Sample_Group, samples_AF)
## construct null model
mod0 <- mod[,1]
## to save time, SVA will be applied to a random selection of 5000 CpG sites
set.seed(20191029)
random.idx <- sample(1:nrow(methylation), 5000)
methylation.sva <- methylation[random.idx,]
## missing methylation values are replaced with mean values
methylation.mean <- rowMeans(methylation.sva, na.rm=T)
idx <- which(is.na(methylation.sva), arr.ind=T)
if (nrow(idx) > 0)
  methylation.sva[idx] <- methylation.mean[idx[,"row"]]
sva.fit <- sva(methylation.sva, mod=mod, mod0=mod0)

design <- cbind(mod, sva.fit$sv)

library(limma)
fit <- lmFit(methylation, design)
fit <- eBayes(fit)

stats <- data.frame(estimate=fit$coefficients[,"Sample_GroupSR"],
                    se=sqrt(fit$s2.post) * fit$stdev.unscaled[,"Sample_GroupSR"],
                    p.value=fit$p.value[,"Sample_GroupSR"])



library(devtools)
install_github("mwsill/IlluminaHumanMethylationEPICv2manifest") 
install_github("mwsill/minfi")
library(minfi)

install("jokergoo/IlluminaHumanMethylationEPICv2manifest") # install EPICv2 array annotation
library(IlluminaHumanMethylationEPICv2anno.20a1.hg38)

data(list="IlluminaHumanMethylationEPICv2anno.20a1.hg38")
data(Locations)
data(Other)
annotation <- cbind(as.data.frame(Locations), as.data.frame(Other))
as.data.frame(Other)$EPICv1_Loci
length(as.data.frame(Other)$EPICv1_Loci)
#rownames(annotation) <- as.data.frame(Other)$EPICv1_Loci
rownames(annotation) <- make.unique(as.data.frame(Other)$EPICv1_Loci)


intersectt <- function (x, y) 
{
  if (is.null(x) || is.null(y)) 
    return(NULL)
  u <- as.vector(x)
  v <- as.vector(y)
  c(u[!duplicated(unclass(u)) & (match(u, v, 0L) > 0L)], v[numeric()])
}

common <- intersectt(rownames(methylation), as.data.frame(Other)$EPICv1_Loci)
#common <- intersect(rownames(methylation), rownames(annotation))
annotation <- annotation[match(common, as.data.frame(Other)$EPICv1_Loci),]
stats <- stats[match(common, rownames(stats)),]
methylation <- methylation[match(common, rownames(methylation)),]

stats <- cbind(stats, annotation)

dmrs <- dmrff(estimate=stats$estimate,
              se=stats$se,
              p.value=stats$p.value,
              methylation=methylation,
              chr=stats$chr,
              pos=stats$pos,
              maxgap=500,
              verbose=T)

write.table(dmrs[, c("chr", "start", "end")], file = "DMRs_Chris_allcandidateregionsbackground.bed", sep = "\t", row.names = FALSE, col.names = FALSE, quote = FALSE)

dmrs_0.05 <- dmrs[which(dmrs$p.adjust < 0.05 & dmrs$n > 1),]
kable(dmrs_0.05[1:10,])

dmrs_0.1 <- dmrs[which(dmrs$p.adjust < 0.1 & dmrs$n > 1),]
kable(dmrs_0.1[1:10,])

dmrs <- dmrs[which(dmrs$p.adjust < 0.05 & dmrs$n > 1),]
kable(dmrs[1:10,])

#List information for each CpG site in these regions
sites <- dmrff.sites(dmrs, stats$chr, stats$pos)
sites <- cbind(sites, stats[sites$site, c("UCSC_RefGene_Name", "estimate", "p.value")])
sites <- cbind(sites, dmr=dmrs[sites$region,c("start","end","z","p.adjust")])


write.csv(dmrs,file="dmrs_dmrff_betanorm_Chris.csv")
write.csv(sites,file="dmr_sites_dmrff_betanorm_Chris.csv")


# Add column with dmr identifier

# Convert csv to bed file
# add it to bed file with unix command
# awk '{print $0 "\tDMR" NR}' input.bed > output.bed

setwd("/Users/adrianrodriguez/Desktop/PhD/AF_DNAme_dataset/Analysis_March24/")
bedfilewithidentifiers <-  as.data.frame(read.table("dmrs_dmrff_betanorm_Chris_DMRs.bed",header = FALSE, sep="\t",stringsAsFactors=FALSE))
bedfilewithidentifiers$V4


bedfilewithidentifiers <-  as.data.frame(read.table("dmrs_dmrff_betanorm_Chris_DMRs.bed",header = FALSE, sep="\t",stringsAsFactors=FALSE))
bedfilewithidentifiers$V4

dmrs$identifier <- bedfilewithidentifiers$V4


dmrs_AFhypometh <- dmrs %>% filter(estimate > 0)

dmrs_AFhypermeth <- dmrs %>% filter(estimate < 0)

dmrs_AFhypermeth <- dmrs_AFhypermeth[, c("chr", "start", "end", "identifier")]
dmrs_AFhypometh <- dmrs_AFhypometh[, c("chr", "start", "end", "identifier")]


dmrs_AFhypermeth_granges <- makeGRangesFromDataFrame(dmrs_AFhypermeth,keep.extra.columns = TRUE)
dmrs_AFhypometh_granges <- makeGRangesFromDataFrame(dmrs_AFhypometh,keep.extra.columns = TRUE)



dmrs_AFhypermeth_granges_df <- data.frame(seqnames=seqnames(dmrs_AFhypermeth_granges),
                                          starts=start(dmrs_AFhypermeth_granges)-1,
                                          ends=end(dmrs_AFhypermeth_granges),
                                          names= mcols(dmrs_AFhypermeth_granges)$identifier,
                                          scores=c(rep(".", length(dmrs_AFhypermeth_granges))),
                                          strands=strand(dmrs_AFhypermeth_granges))

dmrs_AFhypometh_granges_df <- data.frame(seqnames=seqnames(dmrs_AFhypometh_granges),
                                         starts=start(dmrs_AFhypometh_granges)-1,
                                         ends=end(dmrs_AFhypometh_granges),
                                         names=mcols(dmrs_AFhypometh_granges)$identifier,
                                         scores=c(rep(".", length(dmrs_AFhypometh_granges))),
                                         strands=strand(dmrs_AFhypometh_granges))

## The following code outputs the sets of DMRs, as shown in Supplementary Table 7).
write.table(dmrs_AFhypermeth_granges_df, file="dmrs_AFhypermeth_Chris_DMRs.bed", quote=F, sep="\t", row.names=F, col.names=F)
write.table(dmrs_AFhypometh_granges_df, file="dmrs_AFhypometh_Chris_DMRs.bed", quote=F, sep="\t", row.names=F, col.names=F)


####################### ####################### ####################### ####################### ####################### #######################


## Clustering of samples based on beta values of CpGs within DMRs called (ðŸ“Š Figure 6a)

DMRs_Chris <- read.csv("dmrs_dmrff_betanorm_Chris.csv")
DMRs_Chris_toparse <- DMRs_Chris[, c("chr", "start", "end", "DMR")]

all_finaldf <- list()

for (i in 1:nrow(DMRs_Chris_toparse)) {
  chr <- DMRs_Chris_toparse$chr[i]
  start <- DMRs_Chris_toparse$start[i]
  end <- DMRs_Chris_toparse$end[i]
  
  q <- GRanges(seqnames=chr,
               ranges=IRanges(start=start, end=end)) 
  grOut <- subsetByOverlaps(infinium_methylation_epicv2_chr_coord_granges, q)
  grOut
  grOut$CpG
  grOut_df <- as.data.frame(grOut)
  
  data_keep_rows <- grOut$CpG  
  beta_matrix2_data_subset <- beta_matrix2[rownames(beta_matrix2) %in% data_keep_rows, ]
  beta_matrix2_data_subset <- as.data.frame(beta_matrix2_data_subset)
  
  beta_matrix2_data_subset$DMR <- DMRs_Chris_toparse$DMR[i]
  
  all_finaldf[[i]] <- beta_matrix2_data_subset
  
  #all_dfs[[paste("DMR_", i, sep = "")]] <- beta_matrix2_data_subset
  
}

final_df <- do.call(rbind, all_finaldf)

final_df_noDMRcolumn <- subset(final_df, select = -DMR)

#Rearrange columns AF and SR
final_df_noDMRcolumn_rearranged <- final_df_noDMRcolumn[, c("207030420039_R01C01", "207030420039_R02C01", "207030420039_R03C01", "207030420039_R04C01", "207030420039_R05C01", "207030420039_R06C01", "207030420045_R01C01", "207030420045_R02C01", "207030420045_R04C01", "207030420045_R07C01", "207030420045_R08C01", "207030420039_R07C01", "207030420039_R08C01", "207030420045_R03C01", "207030420045_R05C01", "207030420045_R06C01")]

#Rearrange CpGs based on significance of DMR and hyperm/hypo
DMRs_Chris_hyper <- read.table("dmrs_AFhypermeth_Chris_DMRs.bed", header = FALSE, sep = "\t")
DMRs_Chris_hypo <- read.table("dmrs_AFhypometh_Chris_DMRs.bed", header = FALSE, sep = "\t")

DMRs_order <- c(DMRs_Chris_hyper$V4,rev(DMRs_Chris_hypo$V4))

final_df$DMR <- factor(final_df$DMR, levels = DMRs_order)

final_df_ordered <- final_df[order(final_df$DMR), ]
final_df_ordered_noDMRcolumn <- subset(final_df_ordered, select = -DMR)

final_df_ordered_noDMRcolumn_rearranged <- final_df_ordered_noDMRcolumn[, c("207030420039_R01C01", "207030420039_R02C01", "207030420039_R03C01", "207030420039_R04C01", "207030420039_R05C01", "207030420039_R06C01", "207030420045_R01C01", "207030420045_R02C01", "207030420045_R04C01", "207030420045_R07C01", "207030420045_R08C01", "207030420039_R07C01", "207030420039_R08C01", "207030420045_R03C01", "207030420045_R05C01", "207030420045_R06C01")]



#Replace column names
colnames(final_df_ordered_noDMRcolumn_rearranged) <- c("47-AF-RA", "25-AF-RA", "25-AF-LA",
                                                       "31-AF-RA","18-AF-LA","313-AF-LA","47-AF-LA",
                                                       "2231-AF-RA","2231-AF-LA","28-AF-RA",
                                                       "28-AF-LA","45-SR-RA","12b-SR-LA","20-SR-LA","45-SR-LA",
                                                       "12b-SR-RA")


final_df_ordered_noDMRcolumn_rearranged_normZ <- t(scale(t(final_df_ordered_noDMRcolumn_rearranged))) #Z-score normalisation of beta values

column_annotation <- data.frame( Category = c(rep("AF", 11), rep("SR", 5)) )
rownames(column_annotation) <- colnames(final_df_ordered_noDMRcolumn_rearranged_normZ)

row_annotation <- data.frame(DMR = final_df_ordered$DMR)
rownames(row_annotation) <- rownames(final_df_ordered_noDMRcolumn_rearranged_normZ)

#Edit manually for "Colour_for_label"
write.csv(final_df_ordered,file="final_df_ordered.csv")

final_df_ordered_withcol <- read.csv("final_df_ordered.csv")

keys <- final_df_ordered$DMR
values <- final_df_ordered_withcol$Colour_for_label
named_vector <- setNames(values, keys)
unique_named_vector <- named_vector[!duplicated(names(named_vector))]

annotation_colors <- list(
  Category = c(AF = "#F4997F", SR = "#43B9D1"),
  DMR = unique_named_vector
)

#Clustering samples
pheatmap(
  final_df_ordered_noDMRcolumn_rearranged_normZ,                      # Data for heatmap (scaled Î²-values)
  cluster_rows = F,               # Cluster CpGs (rows)
  cluster_cols = T,               # Cluster samples (columns)
  color = colorRampPalette(c("#2E5984", "white", "#FF6961"))(50),  # Color gradient
  main = "Heatmap of Methylation Î²-values",  
  show_rownames = F,       # Do not show CpG names    
  show_colnames = TRUE ,   # Show sample names           
  fontsize_row = 1,
  annotation_col = column_annotation,
  annotation_row = row_annotation,
  annotation_colors = annotation_colors,
)


####################### ####################### ####################### ####################### ####################### #######################

## DMRs annotation
## CpG Islands distribution (ðŸ“Š Figure S6d)

#Downloaded CpG islands annotation from UCSC (http://hgdownload.cse.ucsc.edu/goldenpath/hg38/database/)
setwd("/Users/adrianrodriguez/Desktop/PhD/AF_DNAme_dataset/")
CpG_islands_hg38 <- read.table("cpgIslandExt.txt", header = FALSE, sep = "\t")  

DMRs_Chris_hyper <- read.table("dmrs_AFhypermeth_Chris_DMRs_edited.bed", header = FALSE, sep = "\t")
DMRs_Chris_hypo<- read.table("dmrs_AFhypometh_Chris_DMRs_edited.bed", header = FALSE, sep = "\t")


#Hyper
DMRs_Chris_hyper

gr1 <- GRanges(seqnames = DMRs_Chris_hyper$V1,   # Chromosome
               ranges = IRanges(start = DMRs_Chris_hyper$V2, end = DMRs_Chris_hyper$V3)) 

CpG_islands_hg38_bed <- CpG_islands_hg38[, c("V2", "V3", "V4", "V5")]

gr2 <- GRanges(seqnames = CpG_islands_hg38_bed$V2,   # Chromosome
               ranges = IRanges(start = CpG_islands_hg38_bed$V3, end = CpG_islands_hg38_bed$V4)) 

overlap_indices <- findOverlaps(gr1, gr2)



#Hypo

DMRs_Chris_hypo

gr1 <- GRanges(seqnames = DMRs_Chris_hypo$V1,   # Chromosome
               ranges = IRanges(start = DMRs_Chris_hypo$V2, end = DMRs_Chris_hypo$V3)) 

CpG_islands_hg38_bed <- CpG_islands_hg38[, c("V2", "V3", "V4", "V5")]

gr2 <- GRanges(seqnames = CpG_islands_hg38_bed$V2,   # Chromosome
               ranges = IRanges(start = CpG_islands_hg38_bed$V3, end = CpG_islands_hg38_bed$V4)) 

overlap_indices <- findOverlaps(gr1, gr2)



length(overlap_indices)


#Hyper and hypo 

DMRs_Chris <- rbind(DMRs_Chris_hyper, DMRs_Chris_hypo)

gr1 <- GRanges(seqnames = DMRs_Chris$V1,   # Chromosome
               ranges = IRanges(start = DMRs_Chris$V2, end = DMRs_Chris$V3)) 

CpG_islands_hg38_bed <- CpG_islands_hg38[, c("V2", "V3", "V4", "V5")]

gr2 <- GRanges(seqnames = CpG_islands_hg38_bed$V2,   # Chromosome
               ranges = IRanges(start = CpG_islands_hg38_bed$V3, end = CpG_islands_hg38_bed$V4)) 

overlap_indices <- findOverlaps(gr1, gr2)


overlapping_regions_gr1 <- gr1[queryHits(overlap_indices)]

length(overlap_indices)
query_hits <- queryHits(overlap_indices)

DMRs_overlapping_CGIs <- DMRs_Chris[query_hits,]
DMRs_nonoverlapping_CGIs <- DMRs_Chris[-query_hits,]




Type_of_Region <- c("Within CGIs", "Outside CGIs")
DMRs <- c(length(overlap_indices), nrow(DMRs_Chris) - length(overlap_indices))
result_df <- data.frame(Type_of_Region, DMRs)

ggplot(result_df, aes(x = "", y = DMRs, fill = Type_of_Region)) +
  geom_bar(width = 1, stat = "identity", color = "black", size = 0.5) +
  coord_polar("y", start = 0) +
  theme_void() +
  scale_fill_manual(values = c("#ffaf4b","#BC4E09")) +
  labs(title = "Distribution of DMRs relative to CpG islands") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
                                                                       panel.background = element_blank())





####################### ####################### ####################### ####################### ####################### #######################


## Plot methylation levels in candidate loci of interest
# As shown in ðŸ“Š Figures 6b, 6c, 6d, 7b, 7d and S7a

beta_matrix2 <- norm_noXY.beta
dim(beta_matrix2)


# 1. Load EPIC array hg38 chromosome coordinates annotation
# 2. Extract CpGs from EPIC array wihtin your genomic window of interest (can be an annotated promoter or CpG island, H3K27ac peak of interest from our dataset or an arbitrary genomic window)
# 3. Extract those CpGs from beta values matrix and save data
# 4. Edit data for those CpGs so that we incorporate information such as disease status of samples and plot boxplots
#Represent as a trend line in genomic coordinates !!


# 1. Read EPIC Array data (EPIC-8v2-0_A1.csv)
infinium_methylation_epicv2_chr_coord <- as.data.frame(read.table("infinium_methylation_epicv2_chr_coord2.bed",header = FALSE, sep="\t",stringsAsFactors=FALSE))

dim(infinium_methylation_epicv2_chr_coord)
names(infinium_methylation_epicv2_chr_coord)[names(infinium_methylation_epicv2_chr_coord) == 'V1'] <- 'chr'
names(infinium_methylation_epicv2_chr_coord)[names(infinium_methylation_epicv2_chr_coord) == 'V2'] <- 'start'
names(infinium_methylation_epicv2_chr_coord)[names(infinium_methylation_epicv2_chr_coord) == 'V3'] <- 'end'
names(infinium_methylation_epicv2_chr_coord)[names(infinium_methylation_epicv2_chr_coord) == 'V4'] <- 'CpG'
infinium_methylation_epicv2_chr_coord_granges <- GenomicRanges::makeGRangesFromDataFrame(infinium_methylation_epicv2_chr_coord, keep.extra.columns=TRUE)
infinium_methylation_epicv2_chr_coord_granges
# We have all the EPIC array data with hg38 genomic coordinates
# We want to subset a GRanges object based on a specific window of interest


# 2. Extract CpGs from EPIC array genomic coordinates within a genomic window of interest

## E.g.:for MYLK2 DMR, plot methylation levels for coordinates "chr20:31,819,456-31,819,588".
## Replace values of 'chr', 'start' and 'end' depending on DMR of interest to explore 

chr <- "chr20"
start <-31819456
end <- 31819588

q <- GRanges(seqnames=chr,
             ranges=IRanges(start=start, end=end)) 
grOut <- subsetByOverlaps(infinium_methylation_epicv2_chr_coord_granges, q)
grOut
grOut$CpG
grOut_df <- as.data.frame(grOut)

# 3. Extract beta values data from those CpGs in our dataset
beta_matrix2
data_keep_rows <- grOut$CpG  
beta_matrix2_data_subset <- beta_matrix2[rownames(beta_matrix2) %in% data_keep_rows, ]


# 4. Edit data and plot
beta_matrix2_data_subset_df <- as.data.frame(beta_matrix2_data_subset)
beta_matrix2_data_subset_df$CpG <- rownames(beta_matrix2_data_subset_df)
beta_matrix2_data_subset_df_melted <- melt(as.data.table(beta_matrix2_data_subset_df))



beta_matrix2_data_subset_df_melted_edited <- beta_matrix2_data_subset_df_melted %>%
  mutate(Status_both = case_when(
    startsWith(as.character(variable), "207030420039_R01C01") ~ "AF",
    startsWith(as.character(variable), "207030420039_R02C01") ~ "AF",
    startsWith(as.character(variable), "207030420039_R03C01") ~ "AF",
    startsWith(as.character(variable), "207030420039_R04C01") ~ "AF",
    startsWith(as.character(variable), "207030420039_R05C01") ~ "AF",
    startsWith(as.character(variable), "207030420039_R06C01") ~ "AF",
    startsWith(as.character(variable), "207030420039_R07C01") ~ "SR",
    startsWith(as.character(variable), "207030420039_R08C01") ~ "SR",
    startsWith(as.character(variable), "207030420045_R01C01") ~ "AF",
    startsWith(as.character(variable), "207030420045_R02C01") ~ "AF",
    startsWith(as.character(variable), "207030420045_R03C01") ~ "SR",
    startsWith(as.character(variable), "207030420045_R04C01") ~ "AF",
    startsWith(as.character(variable), "207030420045_R05C01") ~ "SR",
    startsWith(as.character(variable), "207030420045_R06C01") ~ "SR",
    startsWith(as.character(variable), "207030420045_R07C01") ~ "AF",
    startsWith(as.character(variable), "207030420045_R08C01") ~ "AF"
  ))


#Assign genomic coordinate (start CpG position) from grOut_df
grOut_df
grOut_df <- as.data.frame(grOut)
library(dplyr)
beta_matrix2_data_subset_df_melted_edited2 <- left_join(grOut_df,beta_matrix2_data_subset_df_melted_edited, by="CpG")

ggplot(beta_matrix2_data_subset_df_melted_edited2, aes(x = start, y = value, color = Status_both, fill = Status_both)) + 
  geom_point(size = 2, alpha = 1) +  # Keep the dots
  geom_smooth(method = "loess", se = TRUE, alpha = 0.3) +  # Trend lines with shaded error
  scale_x_continuous(limits = c(start, end)) + 
  theme_classic() + 
  scale_color_manual(values = color) + 
  scale_fill_manual(values = color) + 
  xlab("Genomic position") + 
  ylab("DNA methylation (Î²-value)")




####################### ####################### ####################### ####################### ####################### #######################


## Integration of DNAm data with H3K27ac ChIP-seq and RNA-seq DEGs
## Related to Figure S6b


# Genes associated to DMRs assigned based on genomic proximity (GREAT tool; https://great.stanford.edu/great/public/html/)

DMR_Genes <- read.delim("dmrs_dmrff_Chris_associated_Genes.txt", header = TRUE, sep = "")

DMR_Genes_AFhypermeth <- read.delim("dmrs_dmrff_Chris_AFhypermeth_associated_genes.txt", header = TRUE, sep = "")

DMR_Genes_AFhypometh <- read.delim("dmrs_dmrff_Chris_AFhypometh_associated_genes.txt", header = TRUE, sep = "")





